File: C10_Lights/LightSequencerCS/SequencePlayer.cs
//////////////////////////////////////////////////////////////////////////////////
//	SequencePlayer.cs
//	Light Sequencer
//	Written by Brian Peek (http://www.brianpeek.com/)
//	for the Animated Holiday Lights article
//		at Coding4Fun (http://msdn.microsoft.com/coding4fun/)
//////////////////////////////////////////////////////////////////////////////////

using System;

namespace LightSequencer
{
	public abstract class SequencePlayer
	{
		public abstract void Start();
		public abstract void Stop();
		public abstract void Load();
		public abstract void Unload();

		public event EventHandler SequenceStopped;

		protected virtual void OnSequenceStopped(EventArgs e)
		{
			if(SequenceStopped != null)
				SequenceStopped(this, e);
		}
	}
}

File: C10_Lights/LightSequencerVB/SequencePlayer.vb
'////////////////////////////////////////////////////////////////////////////////
'	SequencePlayer.cs
'	Light Sequencer
'	Written by Brian Peek (http://www.brianpeek.com/)
'	for the Animated Holiday Lights article
'		at Coding4Fun (http://msdn.microsoft.com/coding4fun/)
'////////////////////////////////////////////////////////////////////////////////


Imports Microsoft.VisualBasic
Imports System

Namespace LightSequencer
	Public MustInherit Class SequencePlayer
		Public MustOverride Sub Start()
		Public MustOverride Sub [Stop]()
		Public MustOverride Sub Load()
		Public MustOverride Sub Unload()

		Public Event SequenceStopped As EventHandler

		Protected Overridable Sub OnSequenceStopped(ByVal e As EventArgs)
			RaiseEvent SequenceStopped(Me, e)
		End Sub
	End Class
End Namespace
